login-buttons/src/main/java/com/shaishavgandhi/loginbuttons/GoogleButton.java

```diff
@@ -0,0 +1,54 @@
+package com.shaishavgandhi.loginbuttons;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.Typeface;
+import android.util.AttributeSet;
+import android.util.TypedValue;
+
+public final class GoogleButton extends BaseButton {
+    private final Paint linePaint = new Paint();
+
+    public GoogleButton(Context context, AttributeSet attrs) {
+        super(context, attrs, android.R.color.white, R.drawable.google_logo);
+        setDefaults();
+    }
+
+    public GoogleButton(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle, android.R.color.white, R.drawable.google_logo);
+        setDefaults();
+    }
+
+    public GoogleButton(Context context) {
+        super(context);
+        setDefaults();
+    }
+
+    private void setDefaults() {
+        setTextColor(Color.parseColor("#FF586979"));
+        setTextSize(TypedValue.COMPLEX_UNIT_SP, 18);
+        setTypeface(Typeface.DEFAULT);
+
+        linePaint.setColor(Color.parseColor("#EBECED"));
+        linePaint.setStrokeWidth(Utils.convertDpToPixel(1, getContext()));
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        super.onDraw(canvas);
```

Comment:
Could you take a look at the other buttons and how they're implemented? This implementation will hardcode it to a particular look. We want to make it extensible in most looks (rounded corner, transparent, square etc, circular etc)